### PR TITLE
Improve docs & doc examples for HashSet.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: src/libstd/collections/hash/set.rs [hunk 1]
```diff
@@ -146,7 +146,8 @@ impl<T: Hash + Eq> HashSet<T, RandomState> {
     ///
     /// ```
     /// use std::collections::HashSet;
-    /// let mut set: HashSet<i32> = HashSet::with_capacity(10);
+    /// let set: HashSet<i32> = HashSet::with_capacity(10);
+    /// assert!(set.capacity() >= 10);
```
Comment (GuillaumeGomez) [inline src/libstd/collections/hash/set.rs]: Shouldn't it be more accurate to check if the capacity is equal to 10 instead of >= ?

**Comment by GuillaumeGomez** [comment]
Just a small nit. Also, is it normal that you have this much commits?

File: src/libstd/collections/hash/set.rs (hunk 1 above)
Comment (steveklabnik) [inline src/libstd/collections/hash/set.rs]: In general, the allocator may give you more than you asked for.

**Review by steveklabnik — APPROVED** [review]
travis is failing, but overall looks good to me

File: src/libstd/collections/hash/set.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/libstd/collections/hash/set.rs]: Hum ok.

**Comment by QuietMisdreavus** [comment]
Travis says the build was cancelled before it ever started (??) but otherwise the change looks good.

**Comment by frewsxcv** [comment]
rebased, tests pass

@bors r=steveklabnik,quietmisdreavus rollup

**Pull request merged**